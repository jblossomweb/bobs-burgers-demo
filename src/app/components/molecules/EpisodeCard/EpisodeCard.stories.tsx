import React from 'react'
import type { ComponentStory, ComponentMeta } from '@storybook/react'
import mockEpisode from 'app/models/__mocks__/episode.json'

import EpisodeCard, { type Props } from './EpisodeCard'

export default {
  title: 'components/molecules/EpisodeCard',
  component: EpisodeCard,
  argTypes: {
    episode: { control: 'object' },
  },
} as ComponentMeta<typeof EpisodeCard>

const Template: ComponentStory<typeof EpisodeCard> = ({ episode }: Props) => (
  <EpisodeCard episode={episode} />
)

export const Sample = Template.bind({})
Sample.args = {
  episode: mockEpisode,
}
