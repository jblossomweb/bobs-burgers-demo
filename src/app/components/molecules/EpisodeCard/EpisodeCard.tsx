import React from 'react'
import { Link } from 'react-router-dom'
import type Episode from 'app/models/Episode'
import Button from 'app/components/atoms/Button'
import Card from 'app/components/atoms/Card'
import styles from './EpisodeCard.module.css'

export interface Props {
  episode: Episode
}

const EpisodeCard: React.FC<Props> = ({ episode }) => (
  <Card
    className={styles.card}
    width={400}
    title={episode.name}
    foot={
      <Link to={`/episodes/${episode.id}`}>
        <Button className={styles.button}>Details...</Button>
      </Link>
    }
  >
    <p className={styles.info}>{episode.airDate}</p>
  </Card>
)

export default EpisodeCard
