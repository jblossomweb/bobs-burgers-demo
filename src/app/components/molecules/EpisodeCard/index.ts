export { default, type Props } from './EpisodeCard'
