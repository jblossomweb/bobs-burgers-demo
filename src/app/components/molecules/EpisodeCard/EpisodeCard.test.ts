import storySnaps from 'test/utils/storySnaps'
import * as stories from './EpisodeCard.stories'

describe('app/components/molecules/EpisodeCard', () => {
  describe('stories', () => {
    storySnaps(stories)
  })
})
