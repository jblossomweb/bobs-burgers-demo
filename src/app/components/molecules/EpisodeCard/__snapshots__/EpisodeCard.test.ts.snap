// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/components/molecules/EpisodeCard stories Sample matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
    style="width: 400px;"
  >
    <div
      class="card card"
    >
      <div
        class="body"
      >
        <h2
          class="title"
        >
          "Burger War"
        </h2>
        <div
          class="content"
        >
          <p
            class="info"
          >
            April 10, 2011
          </p>
        </div>
      </div>
      <a
        href="/episodes/10"
      >
        <button
          class="button button"
          type="button"
        >
          Details...
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
